pkg/controller/replication/replication_controller.go

```diff
@@ -478,10 +573,19 @@ func (rm *ReplicationManager) manageReplicas(filteredPods []*api.Pod, rc *api.Re
 		var wg sync.WaitGroup
 		wg.Add(diff)
 		glog.V(2).Infof("Too few %q/%q replicas, need %d, creating %d", rc.Namespace, rc.Name, rc.Spec.Replicas, diff)
+		// needs to set the gvk to generate ownerReference
+		oldKind := rc.Kind
+		oldAPIVersion := rc.APIVersion
+		defer func() {
+			rc.Kind = oldKind
+			rc.APIVersion = oldAPIVersion
+		}()
+		rc.APIVersion = v1.SchemeGroupVersion.String()
+		rc.Kind = "ReplicationController"
```

Comment:
??? Is this a hack because we have an internal object here?

At the very least you should get the APIVersion & kind out of the object's self link, that at least we know corresponds to a real object in the server.